### PR TITLE
tests/operator: Refactor "reconcile components" test to use Patch

### PR DESCRIPTION
**Pull request by RamLavi**
### What this PR does
It is common practice on our e2e tests to use Update k8s API in order to
run the test. In order to ensure the change was executed, these also are
wrapped with RetryOnConflict.
In this PR, The RetryOnConflict used on all cases is miss-used so that conflicts
will not be retried (the Get should be inside the RetryOnConflict).

Refactoring test to use Patch, as this would be more precise.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: tests/operator/operator.go [hunk 1]
```diff
@@ -662,23 +661,22 @@ spec:
 			Entry("[test_id:6254] deployments",
 
 				func() {
-
-					vc, err := virtClient.AppsV1().Deployments(originalKv.Namespace).Get(context.Background(), deploymentName, metav1.GetOptions{})
+					patchBytes, err := patch.New(
+						patch.WithAdd(
+							"/spec/template/spec/containers/0/env/-",
```
Comment (codingben) [inline tests/operator/operator.go:666]: What does `/-` by the end string means?

Comment (RamLavi) [reply tests/operator/operator.go:666]: it means "put it at the end of the list"

**Review by codingben — APPROVED** [review]
/lgtm

**Comment by enp0s3** [comment]
/approve
Thank you!

**Pull request merged**